app/src/main/java/de/danoeh/antennapod/fragment/preferences/PlaybackPreferencesFragment.java

```diff
@@ -63,6 +63,20 @@ private void setupPlaybackScreen() {
             behaviour.setEntries(R.array.video_background_behavior_options_without_pip);
             behaviour.setEntryValues(R.array.video_background_behavior_values_without_pip);
         }
+
+        findPreference(UserPreferences.PREF_QUEUE_ADD_TO_FRONT).setOnPreferenceChangeListener(
+                (preference, newValue) -> {
+                    if (newValue instanceof Boolean) {
+                        boolean enableKeepInProgressAtFront = ((Boolean) newValue);
+                        checkKeepInProgressAtFrontItemVisibility(enableKeepInProgressAtFront);
```

Comment:
Why is the setting only selectable if `Enqueue at front` is enabled? What if I press the play button of an episode that is third in queue?

Reply:
The intent of the fix is to Keep in-progress episode at front during enqueuing.
1. If enqueue at front is disabled, enqueuing items would be at the end, so the setting is irrelevant.
2. If user deliberately plays the third episode in the queue, it stays there.

Sounds like you're suggesting a behavior that when enabled, the in-progress (actually currently playing) episode is always at front, i.e., if one plays third episode, it will get moved to the front.

If so, it's also a reasonable behavior. It might be easier for users to understand as well. More logic will be needed and the wording of the settings would be changed.

Thoughts?

Reply:
Couldn't we replace the enqueue switches with something like "Enqueue location": "Front", "Back", "After playing item"? I think that's much more clear than "Keep at front". I, personally, think that "keep" sounds like it does something to the playing queue item. Instead, this PR just decides where to add new items and does not do anything to the currently playing item.

Reply:
How about the following three options?
1. Back
2. Front
3. Front, after in-progress episode

"After playing item" sounds simpler, but it could be somewhat misleading.
If an user deliberately plays 3rd episode on the queue, and autodownload kicks in, with "Keep In-progress Episode at Front" option, the app will enqueue the items at front, rather than after 3rd episode. The option only places the item after the playing one if and only if the playing episode is at very front (the typical case).

Reply:
> The option only places the item after the playing one if and only if the playing episode is at very front

I think this behavior is quite specific to your use-case. What do you think about changing the bahavior to always enqueue after the currently playing episode? That seems to be much more easy to understand and also useful for more users. Additionally, this would not break your use-case because you probably always play the episode that's in the top of the queue.

Reply:
I'm fine with changing it to enqueue after currently playing episode. 

I'd like to first explain the intended behavior before commiting to the change. I feel it addresses a common pitfall of the current enqueue at Front settings.

Assume user enables enqueue at front, the queue is
```
Episode A, playing
Episode B
Episode C
...
```

Auto download kicks in, Episode D is downloaded. The queue would be
```
Episode D
Episode A, currently playing
Episode B
Episode C 
....
```

- When episode A playback completes, the app then advances to Episode B, not the freshly downloaded Episode D. 
- I feel the behavior defeats the purpose of enqueue to front: getting to the newly downloaded episodes sooner.
- The intent of the Keep In-Progress at Front is to fix it, so that Episode D is put after Episode A.

Reply:
To ensure we're on the same page, the settings change would be:
- replace "Enqueue at Front" with a "Enqueue Location"
- "Enqueue Location" has 3 options
  - at Back, the default
  - at Front
  - after current playing episode

Reply:
Yes, I think that's a lot easier to understand than the current implementation.